client/ayon_core/hosts/maya/api/lib.py

```diff
@@ -2159,21 +2151,31 @@ def get_related_sets(node):
            or cmds.getAttr(f"{s}.id") not in ignored
         )
     ]
+    if not sets:
+        return sets
 
     # Exclude deformer sets (`type=2` for `maya.cmds.listSets`)
-    deformer_sets = cmds.listSets(object=node,
-                                  extendToShape=False,
-                                  type=2) or []
-    deformer_sets = set(deformer_sets)  # optimize lookup
-    sets = [s for s in sets if s not in deformer_sets]
+    exclude_sets = cmds.listSets(object=node,
+                                 extendToShape=False,
+                                 type=2) or []
+    exclude_sets = set(exclude_sets)  # optimize lookup
+
+    # Default nodes to ignore
+    exclude_sets.update({"defaultLightSet", "defaultObjectSet"})
+
+    # Filter out the sets to exclude
+    sets = [s for s in sets if s not in exclude_sets]
 
     # Ignore when the set has a specific suffix
-    sets = [s for s in sets if not any(s.endswith(x) for x in ignore_suffices)]
+    ignore_suffices = ("out_SET", "controls_SET", "_INST", "_CON")
+    sets = [s for s in sets if not s.endswith(ignore_suffices)]
+    if not sets:
+        return sets
 
     # Ignore viewport filter view sets (from isolate select and
     # viewports)
+    view_sets = get_isolate_view_sets()
     sets = [s for s in sets if s not in view_sets]
-    sets = [s for s in sets if s not in defaults]
```

Comment:
Can't tell if this is ok, @antirotor can you confirm?

Reply:
The `defaults` filtering is still there - just higher up added into the `exclude_sets` `set`.

Reply:
But https://github.com/ynput/ayon-core/pull/324/files#diff-d3957520925f7a55982d51ee662c9f5432aaa46f67cdff06858b2d3a9575598eR2167 reset the `sets` variable, or?

Reply:
Note the `sets` but the `view_sets`

All the logic is just filtering the `sets` to exclude whatever we don't want.
Or maybe I'm just misreading what you might be pointing out to be a bug - I just don't see it :D